File: testing/corpora/golden_corpus/14_multi_language_modern/modern_crypto.cs
// @ecdat-synthetic-corpus
using System;
using System.Security.Cryptography;

namespace Ecdat.Corpus.Modern
{
    public class ModernCryptoServices
    {
        public static void EncryptPayload(byte[] key, byte[] nonce, byte[] plaintext, byte[] ciphertext, byte[] tag)
        {
            // AES-256-GCM authenticated encryption in .NET
            using (var aesGcm = new AesGcm(key, 16))
            {
                aesGcm.Encrypt(nonce, plaintext, ciphertext, tag);
            }
        }

        public static byte[] HashData(byte[] source)
        {
            // SHA-512 digest in .NET
            using (var sha512 = SHA512.Create())
            {
                return sha512.ComputeHash(source);
            }
        }

        public static ECDsa CreateEllipticCurveSigner()
        {
            // P-384 ECDSA asymmetric key generator in .NET
            return ECDsa.Create(ECCurve.NamedCurves.nistP384);
        }
    }
}

File: testing/corpora/golden_corpus/14_multi_language_modern/modern_crypto.rs
// @ecdat-synthetic-corpus
//! Golden Corpus Fixture: 14_multi_language_modern/modern_crypto.rs
//! Category: 14_multi_language_modern
//! Modern Rust cryptography leveraging `ring` and modern primitives:
//! - ChaCha20Poly1305 AEAD
//! - AES-256-GCM AEAD
//! - Ed25519 digital signatures
//! - SHA-512 cryptographic digest

use ring::aead::{self, BoundKey, Nonce, NonceSequence, OpeningKey, SealingKey, UnboundKey, AES_256_GCM, CHACHA20_POLY1305};
use ring::digest::{self, SHA512};
use ring::signature::{self, Ed25519KeyPair, KeyPair};

pub fn hash_payload(data: &[u8]) -> digest::Digest {
    digest::digest(&SHA512, data)
}

pub fn create_chacha_sealing_key(key_bytes: &[u8]) -> Result<UnboundKey, ring::error::Unspecified> {
    UnboundKey::new(&CHACHA20_POLY1305, key_bytes)
}

pub fn create_aes_gcm_key(key_bytes: &[u8]) -> Result<UnboundKey, ring::error::Unspecified> {
    UnboundKey::new(&AES_256_GCM, key_bytes)
}

pub fn generate_ed25519_keypair(rng: &ring::rand::SystemRandom) -> Result<Ed25519KeyPair, ring::error::Unspecified> {
    let pkcs8 = Ed25519KeyPair::generate_pkcs8(rng)?;
    Ed25519KeyPair::from_pkcs8(pkcs8.as_ref())
}
